compiler/rustc_index_macros/src/lib.rs

```diff
@@ -18,8 +18,19 @@ mod newtype;
 ///   to create/return a value.
 ///
 /// Internally, the index uses a u32, so the index must not exceed
-/// `u32::MAX`. You can also customize things like the `Debug` impl,
-/// what traits are derived, and so forth via the macro.
+/// `u32::MAX`.
+///
+/// The impls provided by default are Clone, Copy, PartialEq, Eq, and Hash.
+///
+/// Accepted attributes for customization:
+/// - #[derive(HashStable_Generic)]/#[derive(HashStable)]: derives
+///   `HashStable`, as normal.
+/// - #[encodable]: derives `Encodable`/`Decodable`.
+/// - #[orderable]: derives `PartialOrd`/`Ord`, plus step-related methods.
+/// - #[debug_format = "Foo({})"]: derives `Debug` with particular output.
+/// - #[max = 0xFFFF_FFFD]: specifies the max value, which allows niche
+///   optimizations. The default max value is 0xFFFF_FF00.
+/// - #[gate_rustc_only]: makes parts of the generated code nightly-only.
```

Comment:
https://github.com/rust-lang-ci/rust/actions/runs/6990882559/job/19020846394#step:26:27071
```
   warning: unresolved link to `encodable`
    --> compiler/rustc_index_macros/src/lib.rs:28:9
     |
  28 | /// - #[encodable]: derives `Encodable`/`Decodable`.
     |         ^^^^^^^^^ no item named `encodable` in scope
     |
     = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
     = note: `#[warn(rustdoc::broken_intra_doc_links)]` on by default
  
  warning: unresolved link to `orderable`
    --> compiler/rustc_index_macros/src/lib.rs:29:9
     |
  29 | /// - #[orderable]: derives `PartialOrd`/`Ord`, plus step-related methods.
     |         ^^^^^^^^^ no item named `orderable` in scope
     |
     = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
  
  warning: unresolved link to `gate_rustc_only`
    --> compiler/rustc_index_macros/src/lib.rs:33:9
     |
  33 | /// - #[gate_rustc_only]: makes parts of the generated code nightly-only.
     |         ^^^^^^^^^^^^^^^ no item named `gate_rustc_only` in scope
     |
     = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
```

Reply:
Should be fixed by #118338.

How did you notice this? I never notice warnings buried deep within CI runs like this...

Reply:
Reading logs from time to time. It's nice to have github action to automate log diffing and grepping to notice interesting stuff.